wrapper/src/main/java/software/amazon/jdbc/PluginServiceImpl.java

```diff
@@ -244,25 +274,47 @@ public synchronized EnumSet<NodeChangeOptions> setCurrentConnection(
       if (!changes.isEmpty()) {
 
         final Connection oldConnection = this.currentConnection;
+        final boolean isInTransaction = this.isInTransaction;
+        this.sessionStateService.begin();
+
+        try {
+          this.currentConnection = connection;
+          this.currentHostSpec = hostSpec;
+
+          this.sessionStateService.applyCurrentSessionState(connection);
+          this.setInTransaction(false);
+
+          if (isInTransaction && PropertyDefinition.ROLLBACK_ON_SWITCH.getBoolean(this.props)) {
```

Comment:
On line 285, `isInTransaction` is set to false. Wouldnt that make the above `if` statement never entered?